### PR TITLE
Reformat and keep with ci

### PR DESCRIPTION
**Pull request by anatawa12**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -42,3 +42,6 @@ jobs:
         with:
           name: proptest-regressions
           path: tests/*.proptest-regressions
+
+      - name: Check format
+        run: cargo fmt --check
```
Comment (NobodyXu) [inline .github/workflows/ci.yml:47]: I also recommend running `clippy`

**Review by NobodyXu — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -42,3 +42,6 @@ jobs:
         with:
           name: proptest-regressions
```
Comment (NobodyXu) [inline .github/workflows/ci.yml:43]: I suggest also run:

```yaml
 - name: Test release build with debug assertions
   run:  cargo test --release
   env:
     CARGO_PROFILE_RELEASE_DEBUG_ASSERTIONS: 1
     CARGO_PROFILE_RELEASE_OVERFLOW_CHECKS: 1
```

and

```yaml
 - name: Test release build with debug assertions
   run:  cargo test --release
```

**Review by NobodyXu — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (NobodyXu) [reply .github/workflows/ci.yml:43]: Sometimes release build could optimize `unsafe` code and find out some assumptions you made that is wrong.

Comment (NobodyXu) [reply .github/workflows/ci.yml:43]: I also recommend using [`cargo-careful`](https://crates.io/crates/cargo-careful) for testing, you can use [taiki-e/install-action](github.com/taiki-e/install-action) to install it.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -45,3 +45,5 @@ jobs:
 
       - name: Check format
         run: cargo fmt --check
+      - name: Clippy
+        run: cargo clippy
```
Comment (NobodyXu) [inline .github/workflows/ci.yml:49]:
```suggestion
        run: cargo clippy --no-deps
```

**Review by NobodyXu — CHANGES_REQUESTED** [review]

**Review by NobodyXu — APPROVED** [review]

**Comment by NobodyXu** [comment]
There's still some warnings though:

```
  --> tests/7zip_compatiblity_test.rs:45:9
   |
45 |     let mut temp_dir = TempDir::new().unwrap();
   |         ----^^^^^^^^
   |         |
   |         help: remove this `mut`
   |
   = note: `#[warn(unused_mut)]` on by default

warning: variable does not need to be mutable
  --> tests/7zip_compatiblity_test.rs:55:9
   |
55 |     let mut seven_zip_process = Command::new(seven_zip)
   |         ----^^^^^^^^^^^^^^^^^
   |         |
   |         help: remove this `mut`
```

**Review by NobodyXu — APPROVED** [review]

**Comment by anatawa12** [comment]
Thank you for quick reviewing again.

**Pull request merged**